frmts/ogcapi/gdalogcapidataset.cpp

```diff
@@ -1052,34 +1145,47 @@ bool OGCAPIDataset::InitWithMapAPI(GDALOpenInfo *poOpenInfo,
                                    double dfYMin, double dfXMax, double dfYMax)
 {
     auto oLinks = oRoot["links"].ToArray();
-    CPLString osPNG_URL;
-    CPLString osJPEG_URL;
+
+    // Key - mime type, Value url
+    std::map<std::string, CPLString> oMapItemUrls;
 
     for (const auto &oLink : oLinks)
     {
         if (oLink["rel"].ToString() ==
                 "http://www.opengis.net/def/rel/ogc/1.0/map" &&
-            oLink["type"].ToString() == "image/png")
+            oLink["type"].IsValid())
         {
-            osPNG_URL = BuildURL(oLink["href"].ToString());
+            oMapItemUrls[oLink["type"].ToString()] =
+                BuildURL(oLink["href"].ToString());
         }
-        else if (oLink["rel"].ToString() ==
-                     "http://www.opengis.net/def/rel/ogc/1.0/map" &&
-                 oLink["type"].ToString() == "image/jpeg")
+        else
         {
-            osJPEG_URL = BuildURL(oLink["href"].ToString());
+            // For lack of additional information assume we are getting some bytes
+            oMapItemUrls["application/octet-stream"] =
+                BuildURL(oLink["href"].ToString());
         }
     }
 
-    CPLString osImageURL =
-        SelectImageURL(poOpenInfo->papszOpenOptions, osPNG_URL, osJPEG_URL);
+    const std::pair<std::string, CPLString> oContentUrlPair =
+        SelectImageURL(poOpenInfo->papszOpenOptions, oMapItemUrls);
+    const std::string osContentType = oContentUrlPair.first;
+    const CPLString osImageURL = oContentUrlPair.second;
+
     if (osImageURL.empty())
     {
         CPLError(CE_Failure, CPLE_AppDefined,
-                 "Cannot find link to PNG or JPEG images");
+                 "Cannot find link to tileset items");
         return false;
     }
-    const int l_nBands = ((osImageURL == osPNG_URL) ? 4 : 3);
+
+    std::unique_ptr<GDALDataset> dataset =
```

Comment:
opening a sample tile has some cost. It would be preferrable if this could be avoided in common cases like JPEG or PNG

Reply:
Make sense. But isn't that only true for PNG (4 bands). JPEG images can have 1 or 3 bands. Is it ok to assume 3?

Reply:
> Is it ok to assume 3?

Yes, I believe underlying code should auto-expand single bland to 3 band

Reply:
I read https://docs.ogc.org/is/20-057/20-057.html as png can have 1, 2, 3, or 4 bands (grayscale, grayscale and alpha, RGB, RGB and alpha. Are all these cases covered in the code?

Reply:
Ok, I added a new `OGCAPIDataset::FigureBands` method that assume 4 for PNG, 3 for JPEG and otherwise attempts to download a tile. If that fails, then it returns 3.

If we want to change the logic, it is now in one place.